AdobeStockAsset/Model/ResourceModel/Keyword/SaveMultiplyAndAssignToAsset.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Model\ResourceModel\Keyword;
+
+use Magento\AdobeStockAssetApi\Api\Data\AssetInterface;
+use Magento\AdobeStockAssetApi\Api\Data\KeywordInterface;
+use Magento\AdobeStockAsset\Model\ResourceModel\Keyword as KeywordResourceModel;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class SaveMultiplyAndAssignToAsset
+ */
+class SaveMultiplyAndAssignToAsset
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var SaveMultiple
+     */
+    private $saveMultiple;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * SaveMultiplyAndAssignToAsset constructor.
+     *
+     * @param ResourceConnection $resourceConnection
+     * @param SaveMultiple       $saveMultiple
+     * @param LoggerInterface    $logger
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        SaveMultiple $saveMultiple,
+        LoggerInterface $logger
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->saveMultiple = $saveMultiple;
+        $this->logger = $logger;
+    }
+
+    /**
+     * Save keywords and assigned to asset.
+     *
+     * @param AssetInterface $asset
+     * @return void
+     *
+     * @throws CouldNotSaveException
+     */
+    public function execute(AssetInterface $asset): void
+    {
+        try {
+            $keywords = $asset->getKeywords();
+            $assignedKeywords = $this->getAlreadyAssignedKeywords($asset);
```

Comment:
I don't think there is any need to know which keywords are already assigned and which are not during the save operation.

The save keyword operation may be as simple as:
`insert ignore into adobe_stock_keyword (keyword) values ('apple');`
`insert ignore into adobe_stock_asset_keyword (keyword_id, asset_id) values (24, 42);`